odoo/addons/base/models/ir_actions.py

```diff
@@ -1186,6 +1188,15 @@ def copy_data(self, default=None):
                 vals['name'] = _('%s (copy)', vals.get('name', ''))
         return vals_list
 
+    def action_open_record_from_server_action(self):
+        return {
+            "type": "ir.actions.act_window",
+            "target": "current",
+            "views": [[False, "form"]],
+            "res_model": self.env.context["res_model"],
```

Comment:
Instead of playing with context values, why does not this implement the action for the `self` record?